### ISSUE TITLE
ML-Benchmark prediction submission: [CorrDiff-TW1]

### ISSUE THREAD

**Issue by ioloi1128** [WCRP-CORDEX/ml-benchmark#13]
### Institution ID

NTNU-CWA

### Institution Name

National Taiwan Normal University/Central Weather Administration

### Emulator identifier (`emulator_id`)

CorrDiff-TW1

### Emulator Description

CorrDiff employs a physics-inspired two-step approach comprising a UNet regression for the conditional mean and a generative diffusion model for residual correction, both utilizing an enhanced UNet architecture based on the Elucidated Diffusion Model (EDM) design with 6 encoder and decoder layers, attention layers at resolution 28. 
•	Training utilizes the Adam optimizer (learning rate 0.0002, 𝛽_1=0.9, 𝛽_2=0.99).
•	MSE loss is used for the regression step (8/16 million steps) and denoising score matching loss for the diffusion step (8/16 million steps).
•	Inference employs a second-order stochastic sampler over 18 steps with noise variance decaying from σ_max=800 to σ_min=0.002.

### Hardware and Training Details

One node with 8 H200 GPUs.
- ~12.5 hr to train regression + diffusion models on 8M samples (i.e., [training duration](https://github.com/NVIDIA/physicsnemo/issues/706#issuecomment-2463504045)=8M).
- ~12.5 hr total for ESD_pseudo_reality dataset; ~25 hr total for Emulator_hist_future dataset.
- ~4.5 minutes per year of predictions, for ensemble=8.

### Stochastic/Probabilistic Output

yes

### Reference URL

https://doi.org/10.21203/rs.3.rs-3673869/v1

### Repository for Reproducibility

https://github.com/NVIDIA/physicsnemo/tree/main/examples/weather/corrdiff

**Comment by bentian** [comment]
> Hardware and Training Details

- One node with 8 H200 GPUs.
- ~12.5 hr to train regression + diffusion models on 8M samples (i.e., [training duration](https://github.com/NVIDIA/physicsnemo/issues/706#issuecomment-2463504045)=8M).
  - ~12.5 hr total for `ESD_pseudo_reality` dataset; ~25 hr total for `Emulator_hist_future` dataset.
- ~4.5 minutes per year of predictions, for ensemble=8.

**Comment by jgonzalezab** [comment]
Yes, for the ALPS domain predictions should be formatted with dimensions (time, x, y). If possible, please also include lat and lon coordinates in the resulting NetCDF file, as these are required for plotting the results. See #12 for an explanation of why data for this domain is spatially indexed using x and y coordinates.

**Comment by ioloi1128** [comment]
Thank you for the reminder. We will check and revise soon.

**Comment by Harvey212** [comment]
Hi, RosieEade. The order of dimension in ALPS domain for our prediction file is time, y, x, member. And the order of dimension in ALPS domain for template file is time, y, x. I'm not sure what is the inconsistency of order in dimension.

**Comment by ioloi1128** [comment]
Please let us know the format of ALPS need to be  (time, y, x, member) or  (time, x, y, member)?

**Comment by jgonzalezab** [comment]
We’ve been processing the data and everything looks good so far! One small thing, it looks like you submitted two sets of predictions (with and without orography), which is fantastic. Would you mind opening a separate issue for the submission that includes orography? Thanks a lot!!

**Comment by ioloi1128** [comment]
Sure, it is added now.
